File: hashmap/2_sum/solution.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)

	for i, num := range nums {
		complement := target - num
		if j, found := m[complement]; found {
			return []int{j, i}
		}
		m[num] = i
	}
	return []int{}
}

func main() {
	nums := []int{-1, 0, 1, 2, 3, 4, 5}
	target := 8
	answer := twoSum(nums, target)
	fmt.Println("Indices of two numbers that sum up to", target, "are:", answer)
}

File: hashmap/2_sum/solution.py
class Solution:
  def twoSum(self, nums, target):
    hashmap = {}
    for i, num in enumerate(nums):
      complement = target - num
      if complement in hashmap:
        return [hashmap[complement], i]
      hashmap[num] = i
    return []
  
  
# Example usage:
if __name__ == "__main__":
  solution = Solution()
  nums = [2, 7, 11, 15]
  target = 9
  result = solution.twoSum(nums, target)
  print(f"Indices of the two numbers that add up to {target}: {result}")